garage/tf/models/gaussian_mlp_model.py

```diff
@@ -0,0 +1,181 @@
+        threshold value, to avoid numerical issues.
+    :param std_hidden_nonlinearity: Nonlinearity for each hidden layer in
+        the std network.
+    :param std_output_nonlinearity: Nonlinearity for output layer in
+        the std network.
+    :param std_parametrization: How the std should be parametrized. There
+        are a few options:
+        - exp: the logarithm of the std will be stored, and applied a
+            exponential transformation
+        - softplus: the std will be computed as log(1+exp(x))
+    :param hidden_nonlinearity: Nonlinearity used for each hidden layer.
+    :param output_nonlinearity: Nonlinearity for the output layer.
+    """
+
+    def __init__(self,
+                 output_dim,
+                 name=None,
+                 hidden_sizes=(32, 32),
+                 hidden_nonlinearity=tf.nn.tanh,
+                 output_nonlinearity=None,
+                 learn_std=True,
+                 adaptive_std=False,
+                 std_share_network=False,
+                 init_std=1.0,
+                 min_std=1e-6,
+                 max_std=None,
+                 std_hidden_sizes=(32, 32),
+                 std_hidden_nonlinearity=tf.nn.tanh,
+                 std_output_nonlinearity=None,
+                 std_parameterization='exp'):
+        # Network parameters
+        super().__init__(name)
+        self._hidden_sizes = hidden_sizes
+        self._output_dim = output_dim
+        self._learn_std = learn_std
+        self._adaptive_std = adaptive_std
+        self._std_share_network = std_share_network
+        self._std_hidden_sizes = std_hidden_sizes
+        self._min_std = min_std
+        self._max_std = max_std
+        self._std_hidden_nonlinearity = std_hidden_nonlinearity
+        self._std_output_nonlinearity = std_output_nonlinearity
+        self._std_parameterization = std_parameterization
+        self._hidden_nonlinearity = hidden_nonlinearity
+        self._output_nonlinearity = output_nonlinearity
+
+        # Tranform std arguments to parameterized space
+        self._init_std_param = None
+        self._min_std_param = None
+        self._max_std_param = None
+        if self._std_parameterization == 'exp':
+            self._init_std_param = np.log(init_std)
+            if min_std:
+                self._min_std_param = np.log(min_std)
+            if max_std:
+                self._max_std_param = np.log(max_std)
+        elif self._std_parameterization == 'softplus':
+            self._init_std_param = np.log(np.exp(init_std) - 1)
+            if min_std:
+                self._min_std_param = np.log(np.exp(min_std) - 1)
+            if max_std:
+                self._max_std_param = np.log(np.exp(max_std) - 1)
+        else:
+            raise NotImplementedError
+
+    def network_output_spec(self):
+        """Network output spec."""
+        return ['sample', 'mean', 'std', 'std_param', 'dist']
+
+    def _build(self, state_input=None, dist=None):
```

Comment:
according to your _build interface, _build implementations may have any number of positional args (*inputs) but no kwargs.

I think this code will function equivalently if you do
```python
def _build(self, state_input, dist):
   # ...
```
If the user forgets to include one or both of these, Python should raise an error which is equivalent to your `assert`s